### PR TITLE
James 1588

### PR DESCRIPTION
**Pull request by rouazana**
Create a project to centralize Cassandra tools

### CODE REVIEW

**Comment by rouazana** [comment]
Strange, on my repo commits are in the good order (project creation fixups just after the referred commit)

File: backends-common/cassandra/pom.xml [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+        <groupId>org.apache.james</groupId>
+        <version>1.8.3-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>apache-james-backends-cassandra</artifactId>
+    <name>Apache James Cassandra backend</name>
+
+    <dependencies>
+        <dependency>
+            <groupId>com.google.guava</groupId>
```
Comment (mbaechler) [inline backends-common/cassandra/pom.xml]: dependencies are supposed to respect our ordering policy

Comment (mbaechler) [inline backends-common/cassandra/pom.xml]: and should be located into build-for-jdk-8 profile

File: backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/CassandraModuleComposite.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    private final ImmutableList<CassandraType> types;
+
+    public CassandraModuleComposite(CassandraModule... modules) {
+        tables = Arrays.stream(modules)
+            .flatMap(module -> module.moduleTables().stream())
+            .collect(Collectors.collectingAndThen(Collectors.toList(), ImmutableList::copyOf));
+        index = Arrays.stream(modules)
+            .flatMap(module -> module.moduleIndex().stream())
+            .collect(Collectors.collectingAndThen(Collectors.toList(), ImmutableList::copyOf));
+        types = Arrays.stream(modules)
```
Comment (mbaechler) [inline backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/CassandraModuleComposite.java]: we definitively should introduce a ImmutableListCollector to use it everywhere.
See http://www.jayway.com/2013/11/12/immutable-list-collector-in-java-8/ for an example

File: backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/CassandraTypesProvider.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+public class CassandraTypesProvider {
+    private final ImmutableMap<String, UserType> userTypes;
+
+    public CassandraTypesProvider(CassandraModule module, Session session) {
+        userTypes = module.moduleTypes()
+            .stream()
+            .collect(Collectors.collectingAndThen(
+                Collectors.toMap(
+                    CassandraType::getName,
+                    type -> session.getCluster()
```
Comment (mbaechler) [inline backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/CassandraTypesProvider.java]: these 4 lines should be extract in a method

File: backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/CassandraTypesProvider.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+import org.apache.james.backends.cassandra.components.CassandraType;
+
+public class CassandraTypesProvider {
+    private final ImmutableMap<String, UserType> userTypes;
+
+    public CassandraTypesProvider(CassandraModule module, Session session) {
+        userTypes = module.moduleTypes()
+            .stream()
+            .collect(Collectors.collectingAndThen(
+                Collectors.toMap(
```
Comment (mbaechler) [inline backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/CassandraTypesProvider.java]: we should write a ImmutableMapCollector (see http://www.jayway.com/2014/09/29/java-8-collector-for-gauvas-linkedhashmultimap/ )

File: backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/CassandraModuleComposite.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
+package org.apache.james.backends.cassandra.init;
```
Comment (aduprat) [inline backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/CassandraModuleComposite.java]: license missing

File: mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/CassandraMailboxModule.java [hunk 6]
```diff
@@ -0,0 +1,174 @@
+
```
Comment (aduprat) [inline mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/CassandraMailboxModule.java]: license missing

File: backends-common/cassandra/pom.xml (hunk 1 above)
Comment (rouazana) [inline backends-common/cassandra/pom.xml]: ok

File: backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/CassandraModuleComposite.java (hunk 2 above)
Comment (rouazana) [inline backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/CassandraModuleComposite.java]: ok

File: backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/CassandraTypesProvider.java (hunk 3 above)
Comment (rouazana) [inline backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/CassandraTypesProvider.java]: ok

File: backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/CassandraTypesProvider.java (hunk 4 above)
Comment (rouazana) [inline backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/CassandraTypesProvider.java]: ok

File: backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/CassandraModuleComposite.java (hunk 5 above)
Comment (rouazana) [inline backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/CassandraModuleComposite.java]: ok

File: mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/CassandraMailboxModule.java (hunk 6 above)
Comment (rouazana) [inline mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/CassandraMailboxModule.java]: ok

**Comment by mbaechler** [comment]
:+1:

**Comment by aduprat** [comment]
:+1:

**Comment by rouazana** [comment]
test this please

**Comment by mbaechler** [comment]
merged

**Pull request closed without merging**